packages/components/src/Slider/Slider.tsx

```diff
@@ -0,0 +1,206 @@
+import React, { useState, useRef, useLayoutEffect } from 'react';
+import styled from 'styled-components';
+import { DndProvider } from 'react-dnd';
+import TouchBackend from 'react-dnd-touch-backend';
+import { MarginProps } from 'styled-system';
+import { ThemeColors } from '@t3n/theme/src/theme/colors/colors';
+import { theme } from '@t3n/theme';
+import {
+  SliderTrackProps,
+  SliderMarkerList,
+  SliderPointer,
+  SliderLabels,
+  DimensionsProps
+} from './SliderElements';
+import SliderDragLayer from './SliderDragLayer';
+
+export type VariantType = 'light' | 'dark';
+
+export interface SliderProps extends MarginProps {
+  initialValue: number;
+  minValue: number;
+  maxValue: number;
+  highlightColor?: ThemeColors & string;
+  labels?: Array<string>;
+  tracks?: Array<SliderTrackProps>;
+  steps?: number;
+  name: string;
+  onChange?: (value: number) => void;
+}
+
+export interface HTMLElementWithOffset extends HTMLElement {
+  offsetWidth: number;
+  offsetHeight: number;
+  offsetLeft: number;
+  offsetTop: number;
+}
+
+export interface HiddenInputProps {
+  name: string;
+  value: any;
+  onChange?: (value: number) => void;
+}
+
+const StyledSlider = styled.div<SliderProps>`
+  position: relative;
+  display: flex;
+  height: auto;
+  margin-left: ${`${theme.space[2]}px`};
```

Comment:
As we spread the `...marginProps` on it we should use the margin function here, right?
```suggestion
  ${margin}
```